### PR TITLE
search: add code monitoring feature flag and basic homepage

### LINKED ISSUE #15283: Code Monitoring: Create feature flag

### PR DESCRIPTION
**Pull request by limitedmage**
Fixes #15283 and #15353

- Create feature flag for code monitoring
- Create basic homepage for code monitoring
- Create routes for code monitoring homepage, conditional on the feature flag being enabled
- Added nav link to code monitoring page when feature flag is on

image

### CODE REVIEW

**Review by rvantonder — APPROVED** [review]
I like the tentative icon

**Comment by limitedmage** [comment]
> I like the tentative icon

The Figma has a different radio tower icon, this is the one in the Material Design Icons that looked the closes to me :) @quinnkeast  let me know if this works or if I should use a different icon.

[Figma for reference:](https://www.figma.com/file/Krh7HoQi0GFxtO2k399ZQ6/RFC-227-%E2%80%93-Code-monitoring-actions-and-notifications?node-id=2160%3A55)
image

**Review by eseliger — APPROVED** [review]
LGTM! I saw there was the discussion this should be enterprise code, so I guess it should be moved, but doesn't need to happen in this PR.

**Review by attfarhan — APPROVED** [review]

**Comment by quinnkeast** [comment]
> > I like the tentative icon
> 
> The Figma has a different radio tower icon, this is the one in the Material Design Icons that looked the closest to me :) @quinnkeast let me know if this works or if I should use a different icon.
> 
> [Figma for reference:](https://www.figma.com/file/Krh7HoQi0GFxtO2k399ZQ6/RFC-227-%E2%80%93-Code-monitoring-actions-and-notifications?node-id=2160%3A55)
> image

Perfect for a placeholder. @AlicjaSuska's going to make a new icon specifically for Code monitoring, but it can be swapped anytime.

**Review by quinnkeast — APPROVED** [review]

File: client/web/src/code-monitoring/CodeMonitoringPage.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                    title={
+                        <>
+                            Code Monitoring{' '}
+                            <sup>
+                                <span className="badge badge-info text-uppercase">Prototype</span>
+                            </sup>
+                        </>
+                    }
+                    icon={VideoInputAntennaIcon}
+                    actions={<></>}
```
[the remark this answers is not in the document]
Comment (lguychard) [reply client/web/src/code-monitoring/CodeMonitoringPage.tsx:36]: [they are](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4e45922b618065355e1f91826de8c09406321c54/-/blob/client/web/src/components/PageHeader.tsx?utm_source=chrome-extension#L7)

**Review by lguychard — APPROVED** [review]

**Comment by limitedmage** [comment]
FYI @quinnkeast  I added a navbar link for Code Monitoring as discussed on Slack, it will only show if the feature flag is enabled.

image

I know the design team is working on figuring out what to do if there are too many items in the navbar; we can go back to this and modify it if necessary.

**Pull request merged**